torch/csrc/jit/tracer.cpp

```diff
@@ -38,35 +48,39 @@ void postRecordTrace(const PreTraceInfo& info,
   }
 }
 
-thread_local ArgumentStash ArgumentStash::stash;
-
-void ArgumentStash::stashIntListElem(const std::string& arg_name, size_t size, size_t idx, const Variable& var) {
-  // TODO: check type?
-  if (!isTracing(var)) return;
-  auto tracing_state = getTracingState({var});
-  auto & list_trace = stash.intlists.emplace(arg_name, size).first->second;
-  JIT_ASSERT(size == list_trace.size());
-  JIT_ASSERT(idx < list_trace.size());
-  JIT_ASSERT(list_trace[idx] == nullptr);
-  list_trace[idx] = getValueTrace(tracing_state, var);
-}
-
 autograd::Variable getSizeOf(const autograd::Variable& var, int64_t dim) {
-  auto tracing_state = getTracingState({var});
+  auto & tracing_state = getTracingState();
   auto & graph = tracing_state->graph;
 
   auto size_var = autograd::make_variable(at::Scalar(var.size(dim)).toTensor());
-  auto* value = getValueTrace(tracing_state, var);
+  auto* value = getValueTrace(var);
   auto* node = graph->create(aten::size, {value})
                     ->i_(attr::dim, dim);
   node->output()->inferTypeFrom(size_var);
   graph->appendNode(node);
-  setValueTrace(tracing_state, size_var, node->output());
+  setValueTrace(size_var, node->output());
 
   return size_var;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// Argument stash
+////////////////////////////////////////////////////////////////////////////////
+thread_local ArgumentStash ArgumentStash::stash;
+
+void ArgumentStash::stashIntListElem(const std::string& arg_name, size_t size, size_t idx, const Variable& var) {
```

Comment:
Do we still need the stasher now that we have a thread local tracer? (Not sure, but wondering.)

Reply:
I could move this function to `TracingState` if you think that would be nicer, but unfortunately we can't get rid of that. It's our way of retaining the information that was present in tensors that got casted into `IntList`s along their way to ATen.